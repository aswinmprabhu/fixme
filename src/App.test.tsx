import { ConnectedRouter } from "connected-react-router";
import { createMemoryHistory } from "history";
import React from "react";
import { Provider } from "react-redux";
import { withRouter } from "react-router";
import { cleanup, render } from "react-testing-library";
import App from "./App";
import store from "./store/createStore";

afterEach(cleanup);

jest.mock("react-ga");

function renderWithRouter(
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) {
  return {
    ...render(
      <Provider store={store}>
        <ConnectedRouter history={history}>{ui}</ConnectedRouter>
      </Provider>
    ),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history
  };
}
test("check if spinner renders correclty", () => {
  const component = renderWithRouter(<App />);
  expect(component).toMatchSnapshot();
});
