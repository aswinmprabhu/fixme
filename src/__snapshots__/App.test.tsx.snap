// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if spinner renders correclty 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
        <div
            class="core-layout__viewport container-fluid"
        >
            <div
                class="loader"
            >
                <ul>
                    <li>
                        <span />
                    </li>
                    <li>
                        <span />
                    </li>
                    <li>
                        <span />
                    </li>
                    <li>
                        <span />
                    </li>
                    <li>
                        <span />
                    </li>
                    <li>
                        <span />
                    </li>
                    <li>
                        <span />
                    </li>
                    <li>
                        <span />
                    </li>
                    <li>
                        <span />
                    </li>
                    <li>
                        <span />
                    </li>
                    <li>
                        <span />
                    </li>
                    <li>
                        <span />
                    </li>
                </ul>
            </div>
        </div>
    </div>
</body>,
  "container": <div>
    <div
        class="core-layout__viewport container-fluid"
    >
        <div
            class="loader"
        >
            <ul>
                <li>
                    <span />
                </li>
                <li>
                    <span />
                </li>
                <li>
                    <span />
                </li>
                <li>
                    <span />
                </li>
                <li>
                    <span />
                </li>
                <li>
                    <span />
                </li>
                <li>
                    <span />
                </li>
                <li>
                    <span />
                </li>
                <li>
                    <span />
                </li>
                <li>
                    <span />
                </li>
                <li>
                    <span />
                </li>
                <li>
                    <span />
                </li>
            </ul>
        </div>
    </div>
</div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "history": Object {
    "action": "POP",
    "block": [Function],
    "canGo": [Function],
    "createHref": [Function],
    "entries": Array [
      Object {
        "hash": "",
        "key": "o093t8",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
    ],
    "go": [Function],
    "goBack": [Function],
    "goForward": [Function],
    "index": 0,
    "length": 1,
    "listen": [Function],
    "location": Object {
      "hash": "",
      "key": "o093t8",
      "pathname": "/",
      "search": "",
      "state": undefined,
    },
    "push": [Function],
    "replace": [Function],
  },
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
